File: src/main/java/ti/proyectoinia/dtos/CertificadoDto.java
package ti.proyectoinia.dtos;

import lombok.Data;
import ti.proyectoinia.business.entities.Certificado.TipoCertificado;

import java.util.Date;

@Data
public class CertificadoDto {
    private Long id;

    // Datos del solicitante
    private String nombreSolicitante;
    private String especie;
    private String cultivar;
    private String categoria;

    // Datos del muestreo
    private String responsableMuestreo;
    private Date fechaMuestreo;
    private String numeroLote;
    private Float pesoKg;
    private Integer numeroEnvases;

    // Datos del laboratorio
    private Date fechaIngresoLaboratorio;
    private Date fechaFinalizacionAnalisis;
    private String numeroMuestra;

    // Datos del certificado
    private String numeroCertificado;
    private TipoCertificado tipoCertificado;
    private Date fechaEmision;
    private byte[] firmante;
    private Date fechaFirma;

    private boolean brassicaContiene;

    private Long reciboId;

    private boolean activo;

    // Resultados de análisis - Pureza
    private Float purezaSemillaPura;
    private Float purezaMateriaInerte;
    private Float purezaOtrasSemillas;
    private Float purezaOtrosCultivos;
    private Float purezaMalezas;
    private String purezaMalezasToleradas;
    private String purezaPeso1000Semillas;
    private String purezaHumedad;
    private String purezaClaseMateriaInerte;
    private String purezaOtrasSemillasDescripcion;

    // Resultados de análisis - DOSN
    private Float dosnGramosAnalizados;
    private Integer dosnMalezasToleranciaCero;
    private Integer dosnMalezasTolerancia;
    private Integer dosnOtrosCultivos;
    private String dosnBrassicaSpp;

    // Resultados de análisis - Germinación
    private Integer germinacionNumeroDias;
    private Float germinacionPlantulasNormales;
    private Float germinacionPlantulasAnormales;
    private Integer germinacionSemillasDuras;
    private Integer germinacionSemillasFrescas;
    private Float germinacionSemillasMuertas;
    private String germinacionSustrato;
    private Integer germinacionTemperatura;
    private String germinacionPreTratamiento;

    private String otrasDeterminaciones;
    private String nombreFirmante;
    private String funcionFirmante;

}

File: frontend/src/models/Certificado.dto.ts
export enum TipoCertificado {
  DEFINITIVO = 'DEFINITIVO',
  PROVISORIO = 'PROVISORIO'
}

export interface CertificadoDto {
  id: number | null;
  nombreSolicitante: string | null;
  especie: string | null;
  cultivar: string | null;
  categoria: string | null;
  responsableMuestreo: string | null;
  fechaMuestreo: string | null; // ISO string
  numeroLote: string | null;
  numeroEnvases: number | null;
  fechaIngresoLaboratorio: string | null; // ISO string
  fechaFinalizacionAnalisis: string | null; // ISO string
  numeroMuestra: string | null;
  numeroCertificado: string | null;
  tipoCertificado: TipoCertificado | null;
  fechaEmision: string | null; // ISO string
  firmante: Uint8Array | number[];
  fechaFirma: string | null; // ISO string
  reciboId: number | null;
  activo: boolean;

  brassicaContiene: boolean;

  nombreFirmante: string | null;
  funcionFirmante: string | null;
  otrasDeterminaciones: string | null;

  // Resultados de análisis - Pureza
  purezaSemillaPura: number | null;
  purezaMateriaInerte: number | null;
  purezaOtrasSemillas: number | null;
  purezaOtrosCultivos: number | null;
  purezaMalezas: number | null;
  purezaMalezasToleradas: string | null;
  purezaPeso1000Semillas: string | null;
  purezaHumedad: string | null;
  purezaClaseMateriaInerte: string | null;
  purezaOtrasSemillasDescripcion: string | null;

  // Resultados de análisis - DOSN
  dosnGramosAnalizados: number | null;
  dosnMalezasToleranciaCero: number | null;
  dosnMalezasTolerancia: number | null;
  dosnOtrosCultivos: number | null;
  dosnBrassicaSpp: number;

  // Resultados de análisis - Germinación
  germinacionNumeroDias: number | null;
  germinacionPlantulasNormales: number | null;
  germinacionPlantulasAnormales: number | null;
  germinacionSemillasDuras: number | null;
  germinacionSemillasFrescas: number | null;
  germinacionSemillasMuertas: number | null;
  germinacionSustrato: string | null;
  germinacionTemperatura: number | null;
  germinacionPreTratamiento: string | null;
}

